app/src/main/java/wishcantw/vocabulazy/activities/player/fragment/PlayerOptionDialogFragment.java

```diff
@@ -74,31 +80,48 @@ public void setOnPlayPrankListener(OnPlayPrankListener listener) {
 
     @Override
     public void onPlayerOptionChanged(int optionID, int mode, View v, int value) {
+
         if (optionID == PlayerOptionView.IDX_OPTION_MODE) {
             // The value is the mode option index that is selected
-            int newMode = value;
-            OptionSettings optionSettings = mPlayerModel.getOptionSettings().get(newMode);
-            mPlayerOptionDialogView.setPlayerOptionModeContent(optionSettings, false);
+            AppPreference.getInstance().setPlayerOptionMode(value);
+
+            mPlayerOptionDialogView.setPlayerOptionModeContent(
+                    mPlayerModel.getPlayerOptionSettings(),
+                    false,
+                    (AppPreference.getInstance().getPlayerVolume() == 1.0f));
         }
+
         // Refresh option settings
         mPlayerModel.updateOptionSettings(optionID, mode, v, value);
-        // TODO : notify the service that option settings has changed
-         optionChanged();
+
+        Intent intent = new Intent(getActivity(), AudioService.class);
+        intent.setAction(AudioService.OPTION_SETTINGS_CHANGED);
```

Comment:
To @allencheng07 
So finally in the commit, we don't have the function about enabling or disabling "sentence voice" in PlayerModel but the layout still with the "sentence voice" remaining. Right?